c7n/policymetrics.py

```diff
@@ -11,62 +11,48 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import argparse
-from dateutil.parser import parse
 from datetime import datetime, timedelta
 import logging
-import os
 
 from c7n.credentials import SessionFactory
-from c7n.policy import load
+from c7n.exceptions import ArgumentError
 from c7n.resources import load_resources
 from c7n.utils import dumps
 
+log = logging.getLogger('custodian.policymetrics')
 
-def setup_parser():
-    parser = argparse.ArgumentParser()
-    parser.add_argument('-c', '--config', required=True)
-    parser.add_argument('--assume', dest="assume_role")
-    parser.add_argument('--profile')
-    parser.add_argument(
-        '--region', default=os.environ.get('AWS_DEFAULT_REGION', 'us-east-1'))
-    parser.add_argument('--start', type=parse)
-    parser.add_argument('--end', type=parse)
-    parser.add_argument('--days', type=int, default=14)
-    parser.add_argument('--period', type=int, default=60*24*24)
-    return parser
 
+def policy_metrics(options, policies):
```

Comment:
to make this library and not cli oriented, we'd want to just pass the policies, and start, end (pre-validated) here, and then return the data. with metrics command doing any printing of results.